lib/Backend/PageAllocatorPool.cpp

```diff
@@ -37,8 +38,10 @@ void PageAllocatorPool::Shutdown()
     AutoCriticalSection autoCS(&cs);
     if (Instance)
     {
-        CloseHandle(Instance->idleCleanupTimer);
-        Instance->RemoveAll();
+        if (Instance->idleCleanupTimer)
+        {
+            CloseHandle(Instance->idleCleanupTimer);
```

Comment:
I mean the function should not re-enter. if it re-enters there's a bug in caller which we should fix. I understand you approach which make sure it is safe but lost the opportunity to catch the bug in caller. I can use your approach and add assert though

---
In reply to: [86002826](https://github.com/Microsoft/ChakraCore/pull/1892#discussion_r86002826) [](ancestors = 86002826,86002065)